### PR TITLE
First OpenAI call to know which tables we need to perform the SQL query

### PR DESCRIPTION
**Pull request by manuelcheveste**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -28,30 +23,39 @@ app.use(
...
 async function executeQuery(query) {
   return new Promise((resolve, reject) => {
+    // Create a new connection pool
+    const pool = new Pool({
+      user: process.env.DB_USER,
+      host: 'localhost',
+      database: process.env.DB_NAME,
+      password: process.env.DB_PASSWORD,
+      port: process.env.DB_PORT,
+    });
```
Comment (diego-suarez) [inline index.js:38]: is this opening the connection on readonly mode? if not, is there a way to specify that connection role?

File: index.js [hunk 2]
```diff
@@ -62,7 +66,40 @@ async function executeQuery(query) {
...
+          You are using Postgres 12.
+          The database tables are: ${tables}
+          Please answer with each name between the char \` .
+          If you think there is a table name that can be used but you aren't sure, please include it anyways.
+        `,
+      },
+      {
+        role: 'user',
+        content: `${queryText}
+        Tell me which tables from the list of tables you would use to make the query.
```
Comment (diego-suarez) [inline index.js]:
```suggestion
        content: `Tell me which tables from the list of tables you would use to make this query:
        ${queryText}`
```

File: index.js [hunk 3]
```diff
@@ -62,7 +66,40 @@ async function executeQuery(query) {
...
+  const completion = await openai.createChatCompletion({
+    temperature: 0.1,
+    model: process.env.OPENAI_CHAT_MODEL,
+    messages: [
+      {
+        role: 'system',
+        content: `You are a helpful assistant that knows a lot about SQL language and manages a database.
+          You are using Postgres 12.
+          The database tables are: ${tables}
+          Please answer with each name between the char \` .
```
Comment (diego-suarez) [inline index.js:86]: Adding an example may be helpful to ensure its not answering with extra text or explanations, like:
```suggestion
          Please answer with each name between the char \` .
          Answer only with a comma separated list of tables, without any explanation. Example response: "`users`, `products`"
```

**Review by diego-suarez — APPROVED** [review]
great! left some suggestions

File: index.js (hunk 1 above)
Comment (manuelcheveste) [reply index.js:38]: I'm not sure if we can specify that on the connection, you can probably chose that when you create the database user for this app and limit the permissions on the SQL operations

**Pull request merged**